src/app/plugins/pubsub.plugin.ts

```diff
@@ -1,23 +1,142 @@
+      })
+      .catch((err) => this.container.loggerService.warn(err));
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    const state = this.container.storeService.get('pubSubStore').state;
-    if (state?.onSale) {
-      message.reply('Chicken tender subs are on sale :eyes:');
-    } else {
-      message.reply('Chicken tender subs are **NOT** on sale.');
+    const input = (args || []).join('-').toLowerCase() || this._DEFAULT_INPUT;
+
+    if (input === 'list' || input === 'types') {
+      //Simply return the list of supported subs
+      await this._generateEmbedList();
+      message.reply(this._EMBED_LIST);
+      return;
+    }
+
+    // checks if their sub was a sub, then if that sub is recognised
+    const subType = this._SUBS.find((sub: string) => sub === input) || 'random';
+
+    //recieves the according info and posts, or derps
+    await this.container.httpService
+      .get(`${this._API_URL}/subs/?name=${subType}`)
+      .then((response: IHttpResponse) => {
+        const [subData] = response.data;
+        const embed: MessageEmbed = this._generateEmbedSub(subData);
+
+        message.reply(embed);
+      })
+      .catch((err) => this.container.loggerService.warn(err));
+  }
+
+  private async _generateEmbedList(): Promise<void> {
+    const lastListUpdate: number = Date.now() - this._LAST_UPD_TIME;
+
+    if (lastListUpdate < this._SUB_UPD_THRESH) {
+      return;
+    }
+
+    this._LAST_UPD_TIME = Date.now(); //since it only updates once a day, don't need to worry about accuracy
+
+    await this._updateList();
+
+    //setting up number of columns and rows for the list
+    const maxCol = 3;
+    const maxRows = 10;
+    const numCols = Math.min(maxCol, Math.ceil(this._SUBS.length / maxRows));
+    const numRows = Math.ceil(this._SUBS.length / numCols);
+
+    //sets up a new embed for the list to reside in
+    this._EMBED_LIST = new MessageEmbed();
+    this._EMBED_LIST.setColor('#0099ff').setTitle('Types of Subs');
+    const subs = this._SUBS.map((sub) => this._normalizeName(sub));
+
+    //cycles through each column inserting their row, and also notes the alphabetic range
+    for (let cols = 0; cols < numCols; cols++) {
+      const columnSubs = subs.slice(numRows * cols, numRows * (cols + 1));
+
+      //each column is a embed field note: only 3 fields can be lined up
+      this._EMBED_LIST.addField(
+        `${columnSubs[0].charAt(0)} - ${columnSubs[columnSubs.length - 1].charAt(0)}`,
+        columnSubs.join('\n'),
+        true
+      );
     }
   }
+
+  private _generateEmbedSub(subData: any): MessageEmbed {
+    const embed: MessageEmbed = new MessageEmbed();
+    const parsedNamed = this._normalizeName(subData.sub_name);
+
+    //defaults to not on sale
+    let saleInfo = `it **isn't** on sale currently :sob:, but who could say no to ${subData.price}.  The last time it was on sale was ${subData.last_sale}`;
```

Comment:
line too long. are you using formatter/linter?

Reply:
yeap, I'll throw in a +.  I was iffy about it myself